server/controllers/authentication/authentication.spec.js

```diff
@@ -0,0 +1,218 @@
+        expect(res.body.error.email[0]).toEqual('The email format is invalid.');
+        done();
+      });
+  });
+});
+
+
+describe('verify account controller', () => {
+  it('should verify user', async (done) => {
+    server()
+      .patch(`/api/v1/auth/verify?token=${token}`)
+      .send(inputObject)
+      .end((err, res) => {
+        expect(res.status).toBe(200);
+        expect(res.body).toHaveProperty('message', 'User email verified successfully');
+        expect(res.body).toHaveProperty('data');
+        done();
+      });
+  });
+});
+
+describe('signup', () => {
+  it('should create a new user', (done) => {
+    server()
+      .post('/api/v1/auth/signup')
+      .send(inputObject)
+      .end((err, res) => {
+        expect(res.status).toBe(200);
+        expect(res.body).toHaveProperty('message', 'success, please go to your inbox to verify your account');
+        expect(res.body).toHaveProperty('data');
+        done();
+      });
+  });
+
+  it('should not create a duplicate user', (done) => {
+    server()
+      .post('/api/v1/auth/signup')
+      .send(inputObject)
+      .end((err, res) => {
+        expect(res.status).toBe(409);
+        expect(res.body).toHaveProperty('message', 'email exists');
+        expect(res.body).toHaveProperty('error');
+        done();
+      });
+  });
+});
+
+describe('login', () => {
+  it('should login', (done) => {
+    server()
+      .post('/api/v1/auth/login')
+      .send(inputObject)
+      .end((err, res) => {
+        expect(res.status).toBe(200);
+        expect(res.body).toHaveProperty('message', 'Authentication successful');
+        expect(res.body).toHaveProperty('data');
+        done();
+      });
+  });
+
+  it('should not login user with credentials that does not exist', (done) => {
+    server()
+      .post('/api/v1/auth/login')
+      .send(getData({ email: 'mosinmiloluwao.owoso@andela.com' }))
+      .end((err, res) => {
+        expect(res.status).toBe(400);
+        expect(res.body).toHaveProperty('message', 'Authentication failed');
+        expect(res.body).toHaveProperty('error');
+        done();
+      });
+  });
+
+  it('should not login user with invalid credentials', (done) => {
+    server()
+      .post('/api/v1/auth/login')
+      .send(getData({ password: '12345678' }))
+      .end((err, res) => {
+        expect(res.status).toBe(400);
+        expect(res.body).toHaveProperty('message', 'Authentication failed');
+        expect(res.body).toHaveProperty('error');
+        done();
+      });
+  });
+});
+
+describe('Authentication Controller Catch Blocks', () => {
+  it('should return server error for signup controller catch block', async () => {
+    const req = { body: { password: 123456 }, sanitizedData: inputObject };
+    const res = {
+      status() {},
+      json() {}
+    };
+
+    sinon.stub(res, 'status').returnsThis();
+    sinon.stub(User, 'create').throws();
+    await signup(req, res);
+    expect(res).toBeTruthy();
```

Comment:
Then here, instead of checking for truthy (cuz it'll always be truthy (^_^)), we can check if it was actually called. 👇 

```suggestion
    expect(res.status).toHaveBeenCalledWith(500);
    // can also assert that `res.json` was called with specific response object
```